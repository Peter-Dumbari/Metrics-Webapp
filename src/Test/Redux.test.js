import axios from 'axios';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import currencyReducer, {
  getCurrency,
  getCurrencyDetails,
  searchCurrency,
} from '../Redux/CurrencySlice';

jest.mock('axios');

const mockStore = configureMockStore([thunk]);

describe('currencySlice', () => {
  let store;

  beforeEach(() => {
    store = mockStore();
  });

  test('fetches currencies and updates state correctly', async () => {
    axios.get.mockResolvedValue({
      data: { USD: 'United States Dollar', EUR: 'Euro', GBP: 'British Pound' },
    });

    await store.dispatch(getCurrency());

    expect(store.getActions()).toMatchSnapshot();
  });

  test('fetches currency details and updates state correctly', async () => {
    const currencyId = 'USD';
    axios.get.mockResolvedValue({ data: { rates: { EUR: 0.85, GBP: 0.74 } } });

    await store.dispatch(getCurrencyDetails(currencyId));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('updates filterCurrency correctly when searching', () => {
    const state = currencyReducer(
      {
        currency: {
          USD: 'United States Dollar',
          EUR: 'Euro',
          GBP: 'British Pound',
        },
        filterCurrency: [],
      },
      searchCurrency('euro'),
    );

    expect(state.filterCurrency).toEqual([['EUR', 'Euro']]);
  });
});
