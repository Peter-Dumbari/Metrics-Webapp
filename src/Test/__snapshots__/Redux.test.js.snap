// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencySlice fetches currencies and updates state correctly 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": undefined,
      "requestId": "YhIdzN9rjMNcexdAspoFg",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "currency/fetchCurrencies/pending",
  },
  Object {
    "error": Object {
      "message": "Cannot read properties of undefined (reading 'data')",
      "name": "TypeError",
      "stack": "TypeError: Cannot read properties of undefined (reading 'data')
    at call (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:16:21)
    at tryCatch (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at Generator._invoke (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at Generator.next (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at asyncGeneratorStep (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at _next (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    },
    "meta": Object {
      "aborted": false,
      "arg": undefined,
      "condition": false,
      "rejectedWithValue": false,
      "requestId": "YhIdzN9rjMNcexdAspoFg",
      "requestStatus": "rejected",
    },
    "payload": undefined,
    "type": "currency/fetchCurrencies/rejected",
  },
]
`;

exports[`currencySlice fetches currency details and updates state correctly 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": "USD",
      "requestId": "uB8V3CZB3GTApa6eDCj51",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "currency/fetchCurrencyDetails/pending",
  },
  Object {
    "error": Object {
      "message": "Cannot read properties of undefined (reading 'data')",
      "name": "TypeError",
      "stack": "TypeError: Cannot read properties of undefined (reading 'data')
    at call (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:24:21)
    at tryCatch (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at Generator._invoke (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at Generator.next (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at asyncGeneratorStep (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at _next (/Users/tams-technology/metrics_webapp/src/Redux/CurrencySlice.js:2:1)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    },
    "meta": Object {
      "aborted": false,
      "arg": "USD",
      "condition": false,
      "rejectedWithValue": false,
      "requestId": "uB8V3CZB3GTApa6eDCj51",
      "requestStatus": "rejected",
    },
    "payload": undefined,
    "type": "currency/fetchCurrencyDetails/rejected",
  },
]
`;
